impl/src/main/java/org/ehcache/internal/store/heap/OnHeapStore.java

```diff
@@ -367,6 +367,16 @@ public void clear() throws CacheAccessException {
 
   @Override
   public void invalidate() throws CacheAccessException {
+    if(map.map != null) {
```

Comment:
The whole invalidation process may take a while. Should we improve this part so that it can happens without interference from other cache operations? I am interested in having other opinions on this.

One idea would be to swap the underlying map while we do the operation as it can be discarded anyway.